bfd/bfd-in2.h

```diff
@@ -3737,6 +3737,8 @@ pc-relative or some form of GOT-indirect relocation.  */
   BFD_RELOC_ARC_S21H_PCREL_PLT,
   BFD_RELOC_ARC_NPS_CMEM16,
   BFD_RELOC_ARC_JLI_SECTOFF,
+  BFD_RELOC_ARC_JLI_SECTOFF_SIMM12,
```

Comment:
As described in the original description of the pull request, these additional relocation entries are used for the additional methods of obtaining the address of a function that is in the JLI table (that do not exist in Metaware right now). These utilize the jli_base AUX register and offset it.

ARC_JLI_32 is for initial values of a global or static variable. Take the following example:
`#pragma jli_call_always(TestFunction)`
`extern void TestFunction(void);`
`static void *gSomeFunction = &TestFunction;`

In this case, during program load, gSomeFunction normally gets set to the address of TestFunction. Instead, with -mjli-func-addr=init this will link to the address of the JLI entry inside the JLI table. Alternatively, the relocation entry could be used during program load to re-write the initial contents as the jli_base AUX register + the section offset for the entry (which is the more preferred formula). This would take into account a move of the JLI table between initial boot and the loading of this program.

ARC_JLI_SECTOFF_SIMM12 is used for the assignment of a function address for a function in the JLI table. Take the following example:
`#pragma jli_call_always(TestFunction)`
`extern void TestFunction(void);`
`void *gSomeFunction;`
`/* some code here */`
`gSomeFunction = &TestFunction;`

At this point, normally gSomeFunction gets the address of TestFunction; however, with -mjli-func-addr=init it will get the address of the entry in the JLI table where the JLI table address is determined by reading the jli_base AUX register (in case it moved). In this case, an lr is followed by a relocated add2 whose immediate value depends on where in the JLI table the enty was linked.